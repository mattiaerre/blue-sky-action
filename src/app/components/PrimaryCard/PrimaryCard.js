import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const PrimaryCard = ({ article }) => (
  <div className="card">
    {article.publishedAt && (
      <div className="card-header">{moment(article.publishedAt).fromNow()}</div>
    )}
    <img src={article.urlToImage} className="card-img-top" alt="" />
    <div className="card-body">
      <h3 className="card-title">
        <a
          href={article.url}
          target="_blank"
          rel="noopener noreferrer"
          dangerouslySetInnerHTML={{ __html: article.title }}
        />
      </h3>
      <p
        className="card-text"
        dangerouslySetInnerHTML={{ __html: article.description }}
      />
    </div>
  </div>
);

PrimaryCard.propTypes = {
  article: PropTypes.object.isRequired
};

export default PrimaryCard;
