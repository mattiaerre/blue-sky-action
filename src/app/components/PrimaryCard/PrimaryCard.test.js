import React from 'react';
import { create } from 'react-test-renderer';
import PrimaryCard from './PrimaryCard';

describe('<PrimaryCard />', () => {
  const props = {
    article: {
      author: 'James Hibberd',
      title:
        '<em>True Detective</em> season 3 casts <em>Fantastic Beasts</em> actress as female lead',
      description: '',
      url: 'http://ew.com/tv/2017/11/17/true-detective-season-3-carmen-ejogo/',
      urlToImage:
        'https://ewedit.files.wordpress.com/2017/11/gettyimages-692777996.jpg?crop=0px%2C0px%2C2700px%2C1417.5px&resize=1200%2C630',
      publishedAt: '2017-11-18T02:30:02Z'
    }
  };
  test.only('to match snapshot', () => {
    const tree = create(<PrimaryCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
