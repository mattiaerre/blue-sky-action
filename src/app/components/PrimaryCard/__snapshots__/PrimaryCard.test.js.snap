// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryCard /> to match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    5 hours ago
  </div>
  <img
    alt=""
    className="card-img-top"
    src="https://ewedit.files.wordpress.com/2017/11/gettyimages-692777996.jpg?crop=0px%2C0px%2C2700px%2C1417.5px&resize=1200%2C630"
  />
  <div
    className="card-body"
  >
    <h3
      className="card-title"
    >
      <a
        dangerouslySetInnerHTML={
          Object {
            "__html": "<em>True Detective</em> season 3 casts <em>Fantastic Beasts</em> actress as female lead",
          }
        }
        href="http://ew.com/tv/2017/11/17/true-detective-season-3-carmen-ejogo/"
        rel="noopener noreferrer"
        target="_blank"
      />
    </h3>
    <p
      className="card-text"
    >
      
    </p>
  </div>
</div>
`;
